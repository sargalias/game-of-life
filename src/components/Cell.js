import React from 'react';

const Cell = (props) => (
  <div
    className={`cell ${props.value === 0 ? 'dead' : 'alive'}`}
    onClick={() => props.onCellClick(props.rowId, props.colId)}
  ></div>
);

export default Cell;
