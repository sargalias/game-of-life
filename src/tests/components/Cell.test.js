import React from 'react';
import { shallow } from 'enzyme';
import Cell from '../../components/Cell';

test('cell with value 0 renders correctly', () => {
  const wrapper = shallow(<Cell value={0} />);
  expect(wrapper).toMatchSnapshot();
});

test('cell with value 1 renders correctly', () => {
  const wrapper = shallow(<Cell value={1} />);
  expect(wrapper).toMatchSnapshot();
});

test('cell calls onCellClick prop when clicked with correct rowId and colId', () => {
  const mock = jest.fn();
  const wrapper = shallow(<Cell rowId={5} colId={3} value={1} onCellClick={mock} />);
  wrapper.simulate('click');
  expect(mock).toHaveBeenCalledTimes(1);
  expect(mock).toHaveBeenLastCalledWith(5, 3);
});

test('cell should render with provided background color if alive', () => {
  const wrapper = shallow(<Cell value={1} bgColor={'blue'} />);
  const html = wrapper.render();
  expect(html.attr('style')).toBe('background-color:blue');
});
