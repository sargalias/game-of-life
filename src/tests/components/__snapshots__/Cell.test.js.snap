// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell with value 0 renders correctly 1`] = `
<div
  className="cell dead"
  onClick={[Function]}
/>
`;

exports[`cell with value 1 renders correctly 1`] = `
<div
  className="cell alive"
  onClick={[Function]}
/>
`;
